lib/api/link_builder.rb

```diff
@@ -0,0 +1,81 @@
+module Api
+  class LinkBuilder
+    PAGING_LINKS = [
+      :self,
+      :next,
+      :previous,
+      :first,
+      :last
+    ].freeze
+
+    attr_reader :offset, :limit, :href, :count, :subcount, :subquery_count
+
+    def initialize(params, href, counts)
+      @offset = params["offset"].to_i if params["offset"]
```

Comment:
Do you need to pass in the `params` if you have what you need in the href arg, and that url needs to be parsed/rebuilt as part of this logic anyway?